src/Framework/Constraint/Count.php

```diff
@@ -22,11 +21,17 @@ class Count extends Constraint
      */
     private $expectedCount;
 
+    /**
+     * @var \SplObjectStorage
+     */
+    private $traversableCounts;
+
     public function __construct(int $expected)
     {
         parent::__construct();
 
-        $this->expectedCount = $expected;
+        $this->expectedCount     = $expected;
+        $this->traversableCounts = new \SplObjectStorage();
```

Comment:
Question: since `$this->traversableCounts` is initialized as needed in `function getCountOfTraversable`, and is never used for the most common case of `array`|`Countable`, would it be preferable to _not_ initialize it here?